[Bug] [Master] Stop dependent node error

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.


### What happened

```
[INFO] 2022-02-09 15:20:57.903 org.apache.dolphinscheduler.server.master.runner.WorkflowExecuteThread:[300] - process event: State Event :key: 241947-0-241947-0 type: PROCESS_STATE_CHANGE executeStatus: READY_STOP task instance id: 0 process instance id: 241947 context: null
[INFO] 2022-02-09 15:20:57.903 org.apache.dolphinscheduler.server.master.runner.WorkflowExecuteThread:[436] - process:241947 state RUNNING_EXECUTION change to READY_STOP
[INFO] 2022-02-09 15:20:57.905 org.apache.dolphinscheduler.server.master.runner.WorkflowExecuteThread:[1273] - kill called on process instance id: 241947, num: 1
[INFO] 2022-02-09 15:20:58.773 org.apache.dolphinscheduler.server.master.processor.queue.TaskResponseService:[223] - already exists handler process size:0
[ERROR] 2022-02-09 15:20:58.787 org.apache.dolphinscheduler.server.master.processor.queue.TaskResponsePersistThread:[77] - persist error, task id:136982, instance id:241947, error: {}
java.lang.NullPointerException: null
        at org.apache.dolphinscheduler.server.master.processor.queue.TaskResponsePersistThread.persist(TaskResponsePersistThread.java:162)
        at org.apache.dolphinscheduler.server.master.processor.queue.TaskResponsePersistThread.run(TaskResponsePersistThread.java:72)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
        at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)
        at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

### What you expected to happen

stop successfully.

### How to reproduce

stop successfully.

### Anything else

_No response_

### Version

2.0.3

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponsePersistThread.java
+++ b/dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponsePersistThread.java
@@ -160,8 +160,8 @@ public class TaskResponsePersistThread implements Runnable {
                 }
 
                 if (channel != null) {
-                    channel.writeAndFlush(taskKillAckCommand.convert2Command());
                     TaskKillAckCommand taskKillAckCommand = new TaskKillAckCommand(ExecutionStatus.SUCCESS.getCode(), taskResponseEvent.getTaskInstanceId());
+                    channel.writeAndFlush(taskKillAckCommand.convert2Command());
                 }
 
                 break;

--- a/dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponsePersistThread.java
+++ b/dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponsePersistThread.java
@@ -74,7 +74,7 @@ public class TaskResponsePersistThread implements Runnable {
                     logger.error("persist meta error, task id:{}, instance id:{}", event.getTaskInstanceId(), event.getProcessInstanceId());
                 }
             } catch (Exception e) {
-                logger.error("persist error, task id:{}, instance id:{}", event.getTaskInstanceId(), event.getProcessInstanceId(), e);
+                logger.error("persist error, task id:{}, instance id:{}, error: {}", event.getTaskInstanceId(), event.getProcessInstanceId(), e);
             } finally {
                 this.events.remove(event);
             }
@@ -159,7 +159,11 @@ public class TaskResponsePersistThread implements Runnable {
                     }
                 }
                 TaskKillAckCommand taskKillAckCommand = new TaskKillAckCommand(ExecutionStatus.SUCCESS.getCode(), taskResponseEvent.getTaskInstanceId());
-                channel.writeAndFlush(taskKillAckCommand.convert2Command());
+
+                if (channel != null) {
+                    channel.writeAndFlush(taskKillAckCommand.convert2Command());
+                }
+
                 break;
             default:
                 throw new IllegalArgumentException("invalid event type : " + event);

--- a/dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponsePersistThread.java
+++ b/dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponsePersistThread.java
@@ -158,10 +158,10 @@ public class TaskResponsePersistThread implements Runnable {
                         logger.debug("ACTION_STOP: task instance id:{}, process instance id:{}", taskResponseEvent.getTaskInstanceId(), taskResponseEvent.getProcessInstanceId());
                     }
                 }
-                TaskKillAckCommand taskKillAckCommand = new TaskKillAckCommand(ExecutionStatus.SUCCESS.getCode(), taskResponseEvent.getTaskInstanceId());
 
                 if (channel != null) {
                     channel.writeAndFlush(taskKillAckCommand.convert2Command());
+                    TaskKillAckCommand taskKillAckCommand = new TaskKillAckCommand(ExecutionStatus.SUCCESS.getCode(), taskResponseEvent.getTaskInstanceId());
                 }
 
                 break;

